simconvoi.cc

```diff
@@ -3418,6 +3425,14 @@ bool can_depart(convoihandle_t cnv, halthandle_t halt, uint32 arrived_time, uint
 	while(  c.is_bound()  ) {
 		const schedule_entry_t e = c->get_schedule()->get_current_entry();
 		coupling_cond |= (e.get_coupling_point()==1  &&  !c->is_coupling_done()  &&  !(c->get_coupling_convoi().is_bound()  &&  c->is_coupled()));
+		if (  c->is_coupling_done()  ||  !c->get_convoi_coupling_in_progress().is_bound()  ||  c->get_convoi_coupling_in_progress()->get_convoi_coupling_in_progress()!=c  ) {
+			// This will_conpling_convoi() flag is too strong. 
+			// So if there are some happens such as
+			// coupling convoi is removed or
+			// forget to remove the flag after coupling done,
+			// this flag should be NONE! 
```

Comment:
```suggestion
			// The convoi_coupling_in_progress flag blocks the departure.
			// Reset the flag if it is outdated to avoid blocking the departure forever.
```